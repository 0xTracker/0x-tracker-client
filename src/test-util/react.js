import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
