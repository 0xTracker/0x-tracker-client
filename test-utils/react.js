import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
