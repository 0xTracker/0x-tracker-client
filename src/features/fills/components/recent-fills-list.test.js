import React from 'react';
import timekeeper from 'timekeeper';

import { renderWithRouter } from '../../../test-util/react';
import RecentFillsList from './recent-fills-list';

beforeAll(() => {
  timekeeper.freeze('2019-11-23T17:57:00Z');
});

afterAll(() => {
  timekeeper.reset();
});

it('should render an assortment of fills', () => {
  const fills = [
    {
      assets: [
        {
          amount: '1',
          price: {
            USD: 29.2429,
          },
          tokenAddress: '0x0e3a2a1f2146d86a604adc220b4967a898d7fe07',
          tokenId: 41011762,
          tokenSymbol: 'CARD',
          tokenType: 'Gods Unchained Cards',
          traderType: 'maker',
          type: 'erc-721',
        },
        {
          amount: '0.19',
          price: {
            USD: 153.91,
          },
          tokenAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
          tokenSymbol: 'WETH',
          tokenType: 'Wrapped Ether',
          traderType: 'taker',
          type: 'erc-20',
        },
      ],
      date: '2019-11-23T17:38:44.000Z',
      feeRecipient: '0x0d056bb17ad4df5593b93a1efc29cb35ba4aa38d',
      id: '5dd96f9d250aa512ae416656',
      makerAddress: '0x5161e1380cd661d7d993c8a3b3e57b059ad8d7a4',
      status: 'successful',
      takerAddress: '0x76481caa104b5f6bccb540dae4cefaf1c398ebea',
      value: {
        USD: 29.2429,
      },
    },
    {
      assets: [
        {
          tokenAddress: '0x0e3a2a1f2146d86a604adc220b4967a898d7fe07',
          tokenId: 122150411,
          traderType: 'maker',
          type: 'erc-721',
        },
        {
          amount: '0.975',
          price: {
            USD: 153.91,
          },
          tokenAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
          tokenSymbol: 'WETH',
          tokenType: 'Wrapped Ether',
          traderType: 'taker',
          type: 'erc-20',
        },
      ],
      date: '2019-11-23T17:38:01.000Z',
      feeRecipient: '0x0d056bb17ad4df5593b93a1efc29cb35ba4aa38d',
      id: '5dd96f6d250aa512ae40bad3',
      makerAddress: '0x55a9c5180dcafc98d99d3f3e4b248e9156b12ac1',
      status: 'successful',
      takerAddress: '0x76481caa104b5f6bccb540dae4cefaf1c398ebea',
      value: {
        USD: 150.06225,
      },
    },
    {
      assets: [
        {
          amount: '0.04030042144',
          price: {
            USD: 153.87,
          },
          tokenAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
          tokenSymbol: 'WETH',
          tokenType: 'Wrapped Ether',
          traderType: 'maker',
          type: 'erc-20',
        },
        {
          amount: '2693',
          price: {
            USD: 0.002302646062745191,
          },
          tokenAddress: '0x503f9794d6a6bb0df8fbb19a2b3e2aeab35339ad',
          tokenSymbol: 'SBT',
          tokenType: 'Star Bit Token',
          traderType: 'taker',
          type: 'erc-20',
        },
      ],
      date: '2019-11-23T17:37:28.000Z',
      feeRecipient: '0x8124071f810d533ff63de61d0c98db99eeb99d64',
      id: '5dd96f3f250aa512ae3fb34a',
      makerAddress: '0x49cb7cedb65fcfeabb4d1dcb24d8da202d370eda',
      status: 'successful',
      takerAddress: '0x0681e844593a051e2882ec897ecd5444efe19ff2',
      value: {
        USD: 6.2010258469728,
      },
    },
    {
      assets: [
        {
          amount: '2693',
          price: {
            USD: 0.0022505749888953757,
          },
          tokenAddress: '0x503f9794d6a6bb0df8fbb19a2b3e2aeab35339ad',
          tokenSymbol: 'SBT',
          tokenType: 'Star Bit Token',
          traderType: 'maker',
          type: 'erc-20',
        },
        {
          tokenAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
          traderType: 'taker',
          type: 'erc-20',
        },
      ],
      date: '2019-11-23T17:37:28.000Z',
      feeRecipient: '0x8124071f810d533ff63de61d0c98db99eeb99d64',
      id: '5dd96f3f250aa512ae3fb34d',
      makerAddress: '0x3997d0f55d1daa549e95c240bc6353636f4cf974',
      status: 'successful',
      takerAddress: '0x0681e844593a051e2882ec897ecd5444efe19ff2',
      value: {
        USD: 6.060798445095247,
      },
    },
  ];
  const { asFragment } = renderWithRouter(<RecentFillsList fills={fills} />);

  expect(asFragment()).toMatchSnapshot();
});
