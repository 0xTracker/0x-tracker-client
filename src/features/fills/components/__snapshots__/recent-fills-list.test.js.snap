// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an assortment of fills 1`] = `
<DocumentFragment>
  <div
    class="recent-fills-list__Wrapper-sc-11wan41-0 bgACHr"
  >
    <div
      class="recent-fills-item__Wrapper-sc-1yc2z6o-0 koJLeh"
    >
      <div
        class="recent-fills-item-image__Wrapper-hf0nj8-0 cAvzIX recent-fills-item___StyledRecentFillsItemImage-sc-1yc2z6o-4 bjoghe"
      >
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper-hf0nj8-3 hyXhcG"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper2-hf0nj8-2 fcpgFi"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
      </div>
      <div
        class="recent-fills-item___StyledDiv-sc-1yc2z6o-5 ljkUpL"
      >
        <h4
          class="recent-fills-item__Heading-sc-1yc2z6o-2 ksELXE"
        >
          <a
            class="link__StyledLink-sc-1m1n17q-0 diqghp"
            href="/fills/5dd96f9d250aa512ae416656"
          >
            1 CARD ⇋ 0.19 WETH
          </a>
        </h4>
        <dl
          class="recent-fills-item__Metadata-sc-1yc2z6o-1 bqxKWz"
        >
          <dt>
            Relayer
          </dt>
          <dd>
            <a
              class="link__StyledLink-sc-1m1n17q-0 diqghp"
              href="/apps/token-trove"
            >
              TokenTrove
            </a>
          </dd>
          <dt>
            Date
          </dt>
          <dd>
            18 minutes ago
          </dd>
        </dl>
      </div>
      <span
        class="recent-fills-item__FillAmount-sc-1yc2z6o-3 dbRRNV"
      >
        $29.24
      </span>
    </div>
    <div
      class="recent-fills-item__Wrapper-sc-1yc2z6o-0 txCrt"
    >
      <div
        class="recent-fills-item-image__Wrapper-hf0nj8-0 cAvzIX recent-fills-item___StyledRecentFillsItemImage-sc-1yc2z6o-4 bjoghe"
      >
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper-hf0nj8-3 hyXhcG"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper2-hf0nj8-2 fcpgFi"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
      </div>
      <div
        class="recent-fills-item___StyledDiv-sc-1yc2z6o-5 ljkUpL"
      >
        <h4
          class="recent-fills-item__Heading-sc-1yc2z6o-2 ksELXE"
        >
          <a
            class="link__StyledLink-sc-1m1n17q-0 diqghp"
            href="/fills/5dd96f6d250aa512ae40bad3"
          >
             Unknown ⇋ 0.975 WETH
          </a>
        </h4>
        <dl
          class="recent-fills-item__Metadata-sc-1yc2z6o-1 bqxKWz"
        >
          <dt>
            Relayer
          </dt>
          <dd>
            <a
              class="link__StyledLink-sc-1m1n17q-0 diqghp"
              href="/apps/token-trove"
            >
              TokenTrove
            </a>
          </dd>
          <dt>
            Date
          </dt>
          <dd>
            19 minutes ago
          </dd>
        </dl>
      </div>
      <span
        class="recent-fills-item__FillAmount-sc-1yc2z6o-3 dbRRNV"
      >
        $150.06
      </span>
    </div>
    <div
      class="recent-fills-item__Wrapper-sc-1yc2z6o-0 koJLeh"
    >
      <div
        class="recent-fills-item-image__Wrapper-hf0nj8-0 cAvzIX recent-fills-item___StyledRecentFillsItemImage-sc-1yc2z6o-4 bjoghe"
      >
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper-hf0nj8-3 hyXhcG"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper2-hf0nj8-2 fcpgFi"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
      </div>
      <div
        class="recent-fills-item___StyledDiv-sc-1yc2z6o-5 ljkUpL"
      >
        <h4
          class="recent-fills-item__Heading-sc-1yc2z6o-2 ksELXE"
        >
          <a
            class="link__StyledLink-sc-1m1n17q-0 diqghp"
            href="/fills/5dd96f3f250aa512ae3fb34a"
          >
            0.0403 WETH ⇋ 2,693 SBT
          </a>
        </h4>
        <dl
          class="recent-fills-item__Metadata-sc-1yc2z6o-1 bqxKWz"
        >
          <dt>
            Relayer
          </dt>
          <dd>
            <a
              class="link__StyledLink-sc-1m1n17q-0 diqghp"
              href="/apps/star-bit"
            >
              STAR BIT
            </a>
          </dd>
          <dt>
            Date
          </dt>
          <dd>
            20 minutes ago
          </dd>
        </dl>
      </div>
      <span
        class="recent-fills-item__FillAmount-sc-1yc2z6o-3 dbRRNV"
      >
        $6.20
      </span>
    </div>
    <div
      class="recent-fills-item__Wrapper-sc-1yc2z6o-0 txCrt"
    >
      <div
        class="recent-fills-item-image__Wrapper-hf0nj8-0 cAvzIX recent-fills-item___StyledRecentFillsItemImage-sc-1yc2z6o-4 bjoghe"
      >
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper-hf0nj8-3 hyXhcG"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
        <div
          class="recent-fills-item-image__ImageWrapper-hf0nj8-1 recent-fills-item-image___StyledImageWrapper2-hf0nj8-2 fcpgFi"
        >
          <svg
            _css2="hsl(235, 14%, 30%)"
            aria-hidden="true"
            class="icon-base__IconBase-ngmsqy-0 hvjdWy fill-assets-image___StyledTokenIcon-sc-1na7yca-2 czOLpO"
            fill="currentColor"
            focusable="false"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-14.243L7.757 12 12 16.243 16.243 12 12 7.757z"
            />
          </svg>
        </div>
      </div>
      <div
        class="recent-fills-item___StyledDiv-sc-1yc2z6o-5 ljkUpL"
      >
        <h4
          class="recent-fills-item__Heading-sc-1yc2z6o-2 ksELXE"
        >
          <a
            class="link__StyledLink-sc-1m1n17q-0 diqghp"
            href="/fills/5dd96f3f250aa512ae3fb34d"
          >
            2,693 SBT ⇋  Unknown
          </a>
        </h4>
        <dl
          class="recent-fills-item__Metadata-sc-1yc2z6o-1 bqxKWz"
        >
          <dt>
            Relayer
          </dt>
          <dd>
            <a
              class="link__StyledLink-sc-1m1n17q-0 diqghp"
              href="/apps/star-bit"
            >
              STAR BIT
            </a>
          </dd>
          <dt>
            Date
          </dt>
          <dd>
            20 minutes ago
          </dd>
        </dl>
      </div>
      <span
        class="recent-fills-item__FillAmount-sc-1yc2z6o-3 dbRRNV"
      >
        $6.06
      </span>
    </div>
  </div>
</DocumentFragment>
`;
